test/jdk/java/util/stream/test/org/openjdk/tests/java/util/stream/SliceOpTest.java

```diff
@@ -356,4 +356,26 @@ public void testLimitParallelHugeInput() {
             assertEquals(LongStream.range(0, n).toArray(), actual);
         }
     }
+
+    public void testSliceOpsSpliteratorPreservesSized() {
+        var parSpliterator = IntStream.range(0, 1000).parallel().skip(50).limit(800).spliterator();
+        assertTrue(parSpliterator.hasCharacteristics(Spliterator.SIZED));
+        assertTrue(parSpliterator.hasCharacteristics(Spliterator.SUBSIZED));
+        assertEquals(parSpliterator.getExactSizeIfKnown(), 800);
+        // Original spliterator is split to [0..499] and [500..999] parts
+        // due to skip+limit, we have [50..499] and [500..849]
+        var prefix = parSpliterator.trySplit();
+        assertNotNull(prefix);
+        assertTrue(parSpliterator.hasCharacteristics(Spliterator.SIZED));
+        assertTrue(parSpliterator.hasCharacteristics(Spliterator.SUBSIZED));
```

Comment:
It would be great to integrate the code comment to the assertion message. Then test failure would print the message making it easier to understand the failure.

It would be great to extract `assertHasCharacteristics` method so the failure printed the actual characteristics rather than "expected true got false"

Reply:
assertTrue/False(spltr.hasCharacteristics(...)) pattern appears in Stream API tests quite often. I can see dozens of occurrences. If extracting such kind of method, then it would be better to replace all of them. At the same time, this will make the patch bigger complicating its review and drifting from the original task. You can contribute such a change separately. In fact, the line number in the report points exactly to the problematic spliterator and characteristics, and dedicated method would not provide you with more information. Anyway, you'll need to take a look at the test source code to see how the spliterator was created, so you won't save much time with a better message.